lib/ansible/module_utils/api.py

```diff
@@ -91,26 +92,57 @@ def ratelimited(*args, **kwargs):
     return wrapper
 
 
-def retry(retries=None, retry_pause=1):
-    """Retry decorator"""
-    def wrapper(f):
+def _retry_limit_exceeded(retries):
+    def function_wrapper(function):
+        def raise_error(*args, **kwargs):
+            raise Exception("Retry limit exceeded: %d" % retries)
```

Comment:
This exception is too broad. It'd be useful to declare a specific error:
```suggestion
            raise RetryError("Retry limit exceeded: %d" % retries)
```

Reply:
I'd rather not modify the existing code unnecessarily, but I'll probably remove the `retry` changes anyway, and hence delete this function to revisit another time in a PR that isn't intended for backport.